### PR TITLE
feat: add mono

### PR DESCRIPTION
**Pull request by Pro**
Specify library name and version:  **mono/6.12.0.122**

Add mono framework to be able to compile .NET sources under linux into conan packages.

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/mono/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+
+required_conan_version = ">=1.33.0"
+
+
+class MonoConan(ConanFile):
+    name = "mono"
+    description = "Mono open source ECMA CLI, C# and .NET implementation."
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://www.mono-project.com/"
+    license = "MIT, BSD"
```
Comment (Croydon) [inline recipes/mono/all/conanfile.py:16]:
```suggestion
    license = ("MIT", "BSD")
```

File: recipes/mono/all/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+            "clang": "6",
+            "apple-clang": "9"
+        }
+
+
+    def validate(self):
+        if self.settings.os == "Windows":
+            raise ConanInvalidConfiguration("mono cannot be built on Windows")
+
+    def configure(self):
```
Comment (Croydon) [inline recipes/mono/all/conanfile.py:85]: All of this should be part of `validate()`

**Comment by Pro** [comment]
@Croydon it seems mono needs to keep the `etc` folder.
Tested removing it, but then the scripts fail, e.g. `cert-sync` fails with this very non-informing exception:

```

Unhandled Exception:
System.TypeInitializationException: The type initializer for 'System.Console' threw an exception. ---> System.TypeInitializationException: The type initializer for 'System.ConsoleDriver' threw an exception. ---> System.TypeInitializationException: The type initializer for 'Sys' threw an exception. ---> System.DllNotFoundException: System.Native assembly:<unknown assembly> type:<unknown type> member:(null)
  at (wrapper managed-to-native) Interop+Sys.LChflagsCanSetHiddenFlag()
  at Interop+Sys..cctor () [0x00000] in <c7ff139eb4ea4e52a759a24a506aba72>:0 
   --- End of inner exception stack trace ---
  at Interop+Sys.Stat (System.ReadOnlySpan`1[T] path, Interop+Sys+FileStatus& output) [0x00028] in <c7ff139eb4ea4e52a759a24a506aba72>:0 
  at System.IO.FileSystem.FileExists (System.ReadOnlySpan`1[T] fullPath, System.Int32 fileType, Interop+ErrorInfo& errorInfo) [0x00007] in <c7ff139eb4ea4e52a759a24a506aba72>:0 
  at System.IO.FileSystem.DirectoryExists (System.ReadOnlySpan`1[T] fullPath, Interop+ErrorInfo& errorInfo) [0x00000] in <c7ff139eb4ea4e52a759a24a506aba72>:0 
  at System.IO.FileSystem.DirectoryExists (System.ReadOnlySpan`1[T] fullPath) [0x00000] in <c7ff139eb4ea4e52a759a24a506aba72>:0 
  at System.IO.Directory.Exists (System.String path) [0x0001e] in <c7ff139eb4ea4e52a759a24a506aba72>:0 
  at System.TermInfoDriver.SearchTerminfo (System.String term) [0x00044] in <c7ff139eb4ea4e52a759a24a506aba72>:0 
  at System.TermInfoDriver..ctor (System.String term) [0x0004b] in <c7ff139eb4ea4e52a759a24a506aba72>:0 
  at System.ConsoleDriver.CreateTermInfoDriver (System.String term) [0x00000] in <c7ff139eb4ea4e52a759a24a506aba72>:0 
  at System.ConsoleDriver..cctor () [0x0004d] in <c7ff139eb4ea4e52a759a24a506aba72>:0 
   --- End of inner exception stack trace ---
  at System.Console.SetupStreams (System.Text.Encoding inputEncoding, System.Text.Encoding outputEncoding) [0x00007] in <c7ff139eb4ea4e52a759a24a506aba72>:0 
  at System.Console..cctor () [0x0007d] in <c7ff139eb4ea4e52a759a24a506aba72>:0 
   --- End of inner exception stack trace ---
  at Mono.Tools.CertSync.Main (System.String[] args) [0x0002b] in <eb68251c8300458e9f876a6572a83efa>:0 
[ERROR] FATAL UNHANDLED EXCEPTION: System.TypeInitializationException: The type initializer for 'System.Console' threw an exception. ---> System.TypeInitializationException: The type initializer for 'System.ConsoleDriver' threw an exception. ---> System.TypeInitializationException: The type initializer for 'Sys' threw an exception. ---> System.DllNotFoundException: System.Native assembly:<unknown assembly> type:<unknown type> member:(null)
  at (wrapper managed-to-native) Interop+Sys.LChflagsCanSetHiddenFlag()
  at Interop+Sys..cctor () [0x00000] in <c7ff139eb4ea4e52a759a24a506aba72>:0 
   --- End of inner exception stack trace ---
  at Interop+Sys.Stat (System.ReadOnlySpan`1[T] path, Interop+Sys+FileStatus& output) [0x00028] in <c7ff139eb4ea4e52a759a24a506aba72>:0 
  at System.IO.FileSystem.FileExists (System.ReadOnlySpan`1[T] fullPath, System.Int32 fileType, Interop+ErrorInfo& errorInfo) [0x00007] in <c7ff139eb4ea4e52a759a24a506aba72>:0 
  at System.IO.FileSystem.DirectoryExists (System.ReadOnlySpan`1[T] fullPath, Interop+ErrorInfo& errorInfo) [0x00000] in <c7ff139eb4ea4e52a759a24a506aba72>:0 
  at System.IO.FileSystem.DirectoryExists (System.ReadOnlySpan`1[T] fullPath) [0x00000] in <c7ff139eb4ea4e52a759a24a506aba72>:0 
  at System.IO.Directory.Exists (System.String path) [0x0001e] in <c7ff139eb4ea4e52a759a24a506aba72>:0 
  at System.TermInfoDriver.SearchTerminfo (System.String term) [0x00044] in <c7ff139eb4ea4e52a759a24a506aba72>:0 
  at System.TermInfoDriver..ctor (System.String term) [0x0004b] in <c7ff139eb4ea4e52a759a24a506aba72>:0 
  at System.ConsoleDriver.CreateTermInfoDriver (System.String term) [0x00000] in <c7ff139eb4ea4e52a759a24a506aba72>:0 
  at System.ConsoleDriver..cctor () [0x0004d] in <c7ff139eb4ea4e52a759a24a506aba72>:0 
   --- End of inner exception stack trace ---
  at System.Console.SetupStreams (System.Text.Encoding inputEncoding, System.Text.Encoding outputEncoding) [0x00007] in <c7ff139eb4ea4e52a759a24a506aba72>:0 
  at System.Console..cctor () [0x0007d] in <c7ff139eb4ea4e52a759a24a506aba72>:0 
   --- End of inner exception stack trace ---
  at Mono.Tools.CertSync.Main (System.String[] args) [0x0002b] in <eb68251c8300458e9f876a6572a83efa>:0 
```

If I keep the etc folder, all works as expected. Should I add an exception here?
https://github.com/conan-io/hooks/blob/b545cbd5c5de5dae9cd71a55c851303748b7954d/hooks/conan-center.py#L906

**Comment by Croydon** [comment]
> If I keep the etc folder, all works as expected. Should I add an exception here?
https://github.com/conan-io/hooks/blob/b545cbd5c5de5dae9cd71a55c851303748b7954d/hooks/conan-center.py#L906

Sounds fine to me

File: recipes/mono/all/conanfile.py [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+class MonoConan(ConanFile):
+    name = "mono"
+    description = "Mono open source ECMA CLI, C# and .NET implementation."
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://www.mono-project.com/"
+    license = ("MIT", "BSD-3-Clause", "Apache-2.0")
+    exports_sources = "patches/**"
+    settings = "os", "compiler"
+    topics = "dotnet", "mono", "csharp", "runtime", "compiler"
+    # options = {
```
Comment (uilianries) [inline recipes/mono/all/conanfile.py:20]: what about these options? Can we include them? Otherwise, please, remove

**Pull request closed without merging**